### ISSUE TITLE
Warnings: non-transient instance field of a serializable class declared with a non-serializable type

### ISSUE THREAD

**Issue by milleruntime** [apache/accumulo#2776]
<pre>
java: non-transient instance field of a serializable class declared with a non-serializable type
</pre>
These warnings will need to be fixed. Building with Java 18 showed the warnings in these classes:

- [x] ByteSequence.java
- [x]  MutationsRejectedException.java
- [ ] TableInfo.java
- [ ] CloneInfo.java
- [ ] NamespaceInfo.java
- [ ] ImportedTableInfo.java
- [x] TimedOutException
- [x] TabletLocationState

These are all `Serializable` classes that have non-serializable types. For instance:
https://github.com/apache/accumulo/blob/cf57e343de77f416de10c22e0ed05ef37f433b36/core/src/main/java/org/apache/accumulo/core/security/Authorizations.java#L47-L48

Edit: Merged #2777 so added ByteSequence

**Comment by milleruntime** [comment]
Related PR https://github.com/apache/accumulo/pull/2777

**Comment by ctubbsii** [comment]
This is just a warning... what's the real-world impact? Are these effectively not serializable? Like, if we try, they could never have been serialized? If they have never been serializable, then we're not breaking any user code by dropping it... we're just breaking binary compatibility, which we don't guarantee across minor/major versions anyway, and we can just drop the useless interface in 3.0.

If they are still serializable as a whole, but behaves in a weird way due to the fields not being serializable, then we may need to try to preserve existing weird behavior or try to change behavior through a deprecation cycle.

**Comment by ctubbsii** [comment]
> > we're just breaking binary compatibility, which we don't guarantee across minor/major versions anyway
> 
> The public API definition says, "The Accumulo project maintains binary compatibility across this API within a major release...", which I take to mean binary compatibility across all 2.x releases. Am I reading that wrong?
> 
> I'm thinking we should address this in a 3.0 when we are making other API changes.

You're right. We do try to preserve binary compatibility for minor versions. I didn't recall correctly. I think this is one thing we can relax, when needed, but it's a good goal to at least attempt to preserve binary compatibility.

**Comment by milleruntime** [comment]
> I'm not sure that these warning indicate an error will occur...

Right. They are just warnings.

> I'm thinking we should address this in a 3.0 when we are making other API changes.

If we can fix the warnings without touching the API like I did in #2777, then I think that would be good to get a head start.

**Comment by dlmarion** [comment]
> If we can fix the warnings without touching the API like I did in https://github.com/apache/accumulo/pull/2777, then I think that would be good to get a head start.

Yeah, I'm all for that.
